src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -407,6 +419,33 @@
 )
 
 
+@snowflake_dialect.segment()
+class ConnectByClauseSegment(BaseSegment):
+    """A `CONNECT BY` clause.
+
+    https://docs.snowflake.com/en/sql-reference/constructs/connect-by.html
+    """
+
+    type = "connectby_clause"
+    match_grammar = StartsWith(Sequence("START", "WITH"))
+    parse_grammar = Sequence(
+        "START",
+        "WITH",
```

Comment:
This doesn't add anything and can be replaced with this:

```suggestion
    match_grammar = Sequence(
        "START",
        "WITH",
```